### DISCUSSION TITLE
zfs_autobackup created file systems report `busy` to attempts to destroy -r

### DISCUSSION THREAD

**Discussion by hput** [psy0rz/zfs_autobackup#126] [Q&A] [answered]
After having tested zfs_autobackup; followed the README and `get started'. I find the filesystem put in place by zfs_autobackup; all
report `busy' if I try to destroy them.   The test results live on a VBox vm built as Backup host. That host has been rebooted more than once since zfs_autobackup put the filesystems in place.

These `busy' systems, in accord with the tutorial here for zfs_autobackup were marked with "zfs set  test1".

What might be causing them to show busy, and why.

Results of attempt to destroy: 
zfs destroy -r p0/bk/zroot/projects cannot destroy snapshot p0/bk/zroot/projects@test1-20220426165230: dataset is busy

**Comment by Scrin** [comment, marked as answer]
zfs will show datasets as busy if they have an active hold and you try to delete them. zfs_autobackup uses holds to prevent deleting common snapshots required for incremental transfers.

You can use `zfs holds path/to/dataset@name-of-the-snapshot` to display the holds for a specific snapshot, and `zfs release <tag> path/to/dataset@name-of-the-snapshot` to release such hold. However do note that if you delete the last common snapshot (as marked by the hold), you will be unable to run zfs_autobackup on that dataset without first fully destroying the target dataset.

**Reply by hput** [reply]
Thank you sir for full and complete answer.   I've been pulling my hair out (of a bald head) trying to find a place for questions and discussion on zfs_autobackup.  fs's now destroyed as needed.

The `hold' business seems a practical thing to do, now you've explained it.  Did I miss a heads up in the `start here' docs or README about this hold being a factor under certain circumstances?
Had I more experience I probably would have thought to check for holds.
Thanks again for the clear excellent answer.

**Reply by psy0rz** [reply]
indeed i might need to add a note somewhere about holds

**Reply by psy0rz** [reply]
added it to tips as well: https://github.com/psy0rz/zfs_autobackup/wiki#tips

so now its at 3 places including this forum :)

**Reply by psy0rz** [reply]
Because of another question i decided to write a wiki page for this: https://github.com/psy0rz/zfs_autobackup/wiki/Common-snapshots-and-holds

**Comment by psy0rz** [comment]
Thats correct. I also updated the wiki with this:

https://github.com/psy0rz/zfs_autobackup/wiki/Problems#dataset-is-busy
